### ISSUE TITLE
mypy finds an error ONLY via pre-commit

### ISSUE THREAD

**Issue by WhyNotHugo** [pre-commit/mirrors-mypy#49]
I'm having a really weird issue where `mypy` detects an error _only_ if run via pre commit.

```
$ mypy todoman/formatters.py
Success: no issues found in 1 source file
$ mypy --version
mypy 0.812
```

I also use neovim+efm to run mypy on my source, and that too reports all green.

However:

```
$ pre-commit run --all
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Toml...............................................................Passed
Check for added large files..............................................Passed
Debug Statements (Python)................................................Passed
Reorder python imports...................................................Passed
black....................................................................Passed
flake8...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

todoman/formatters.py:77: error: Argument 1 to "style" has incompatible type "Union[str, int, None]"; expected "str"
Found 1 error in 1 file (checked 24 source files)
```

The relevant files are here:

https://github.com/pimutils/todoman/pull/438/files#diff-d8c6e7d9d914526d66fa906f33739931987eb5927f1b52a1a4ebe1ee084f3879

Do you have _any_ idea of what might be up? Even installing everything into a virtualenv, I still get no error locally. What could `pre-commit` be doing different?

**Comment by asottile** [comment]
I cannot reproduce your findings:

```bash
#!/usr/bin/env bash
set -euxo pipefail
rm -rf venv .mypy_cache
virtualenv venv -qq
venv/bin/pip install -qq mypy==0.812  # from .pre-commit-config.yaml
venv/bin/mypy --scripts-are-modules --ignore-missing-imports $(git ls-files -- "*.py")
```

```console
$ bash t.sh 
+ rm -rf venv .mypy_cache
+ virtualenv venv -qq
+ venv/bin/pip install -qq mypy==0.812
++ git ls-files -- '*.py'
+ venv/bin/mypy --scripts-are-modules --ignore-missing-imports docs/source/conf.py setup.py tests/conftest.py tests/helpers.py tests/test_backend.py tests/test_cli.py tests/test_config.py tests/test_filtering.py tests/test_formatter.py tests/test_main.py tests/test_model.py tests/test_porcelain.py tests/test_ui.py tests/test_widgets.py todoman/__init__.py todoman/__main__.py todoman/cli.py todoman/configuration.py todoman/exceptions.py todoman/formatters.py todoman/interactive.py todoman/model.py todoman/widgets.py
todoman/__init__.py:1: error: Module 'todoman' has no attribute 'version'; maybe "__version__"?
todoman/formatters.py:77: error: Argument 1 to "style" has incompatible type "Union[str, int, None]"; expected "str"
Found 2 errors in 2 files (checked 23 source files)
```

**Comment by WhyNotHugo** [comment]
I've been able to reproduce the failure inside docker, so it seems that it's not pre-commit that's weird, but rather **all** my development environments.

Thanks for your patience here (and thanks for pre-commit!).
